.github/workflows/setup.yml

```diff
@@ -221,10 +221,11 @@ jobs:
         run: |
          python3 -m venv env
          source env/bin/activate
+         pip install psutil==5.8.0
```

Comment:
This requirement is already defined at `requirements-dev.txt` and if we need to change this is in the future we will chave to update twice.  This file has also other requirements need for selftests.

What about calling `make develop`? Or if installing `make` it is not an option, maybe calling from pip with: `pip install -r requirements-dev.txt`?

Reply:
Hi @beraldoleal

> This requirement is already defined at `requirements-dev.txt` and if we need to change this is in the future we will chave to update twice. This file has also other requirements need for selftests.

This was deliberate I wanted to test with the minimum requeriments possible. This test doesn't run the static checks, so the only requeriment is `psutils`. I agree the fixed version  isn't ideal so maybe the alternative is just to leave it without version? psutils doesn't change that often.

Reply:
Right, you are skipping `static-checks`, but we are running functional tests and unit tests here as well. So, if `psutils` is the only missing dependency, I can think about the following possibilities:

1. Other dependencies are already installed by the system;
3. Some might not be used anymore (i.e: maybe `elementpath`);
3. In some cases, if some dependencies are not installed, then some tests are being skipped (for instance `netifaces` is defined at `requirements-dev.txt` and used at `selftests/functional/plugin/test_vmimage.py`).

So, we are running `python setup.py test`, but not testing everything.

If we would like to have a minimal requirement (and allow some tests to skip), so maybe we could use better the extra_requires feature of pip? and having a default minimal requirement file with an extra called "full" that will install everything.

(I'm not concerned only about the version here. But having the requirements around multiple files it is odd for me.)

Reply:
@beraldoleal ok, thanks for expanding your concerns. I'll replace the step with `pip install -r requirements-dev.txt`